scripts/alert_emails/send_emails.ts

```diff
@@ -78,42 +82,55 @@ function generateSendData(usersToEmail: string[], alertForLocation: Alert) : Ema
     const auth = { apiKey: process.env.CREATE_SEND_TOKEN };
     const api: any = new createsend(auth);
-    const err_results: CampaignMonitorError[] = []
+    let emailSent = 0;
+    const uniqueEmailAddress: {[email: string]: number} = {}
+    const locationsWithEmails: {[fips: string]: number} = {}
 
-    const db = getFirebase().firestore();
+    const db = getFirestore();
     Object.keys(locationsWithAlerts).map(async (fips) => {
         db.collection(`snapshots/${currentSnapshot}/locations/${fips}/emails/`)
             .where("sentAt", "==", null).get().then(querySnapshot => {
                 querySnapshot.forEach(async doc => {
-                    await sendEmail(api, generateSendData([doc.id], locationsWithAlerts[fips]))
+                    await sendEmail(api, generateSendData([doc.id], locationsWithAlerts[fips]), dryRun)
                         .then(async result => {
-                            console.log(result)
-                            await db.collection(`snapshots/${currentSnapshot}/locations/${fips}/emails/`).doc(doc.id).set({sentAt: firebase.firestore.FieldValue.serverTimestamp()})
+                            emailSent += 1;
+                            uniqueEmailAddress[doc.id] = (uniqueEmailAddress[doc.id] || 0) + 1;
+                            locationsWithEmails[fips] = (locationsWithEmails[fips] || 0) + 1;
+                            if (dryRun) return;
+                            await db.collection(`snapshots/${currentSnapshot}/locations/${fips}/emails/`).doc(doc.id)
+                            .set({
+                                sentAt: admin.firestore.FieldValue.serverTimestamp()
+                            })
                         }).catch(err => {
                             console.log(err);
-                            err_results.push(err) // more elegantly handle the errors here?
+                            process.exit(-1);
                         });
                 })
             });
         });
-
+    console.log(`Total Emails to be sent: ${emailSent}. Total locations with emails ${locationsWithEmails}. Unique Email addresses ${Object(locationsWithEmails).keys.length}`)
     console.log(`Done.`);
 
 })();
 
-async function parseArgs(): Promise<{ fipsToAlertFilename: string, currentSnapshot: string }> {
+async function parseArgs(): Promise<{ fipsToAlertFilename: string, currentSnapshot: string, dryRun: boolean }> {
     const args = process.argv.slice(2);
     if (args.length == 2) {
         const fipsToAlertFilename = args[0];
         const currentSnapshot = args[1];
-        return { fipsToAlertFilename, currentSnapshot };
-    } else {
+        return { fipsToAlertFilename, currentSnapshot, dryRun: true };
+    } if (args.length == 3) {
+        const fipsToAlertFilename = args[0];
+        const currentSnapshot = args[1];
+        const isSend = args[2] === "true";
+        return { fipsToAlertFilename, currentSnapshot, dryRun: !isSend};
+    }else {
```

Comment:
hrm. Somehow this isn't prettier'ed!

Reply:
ya actually the linter hasn't really been working for me recently :/